docs/source/ProgrammingGuide/Initialization.rst

```diff
@@ -0,0 +1,142 @@
+
+When passing a boolean as a string, the acceptable values are:
+ - true, yes, 1
+ - false, no, 0
+
+The values are case insensitive.
+
+
+:sup:`1` This is the preferred set of defaults when CUDA and OpenMP are enabled. If you use a thread-parallel host execution space, we prefer Kokkos' OpenMP backend, as this ensures compatibility of Kokkos' threads with the application's direct use of OpenMP threads. Kokkos cannot promise that its Threads backend will not conflict with the application's direct use of operating system threads.
+:sup:`2` The two device-id mapping strategies are:
+- random: select a random device from available.
+- mpi_rank: select device based on a round robin assignment of local MPI ranks. Works with OpenMPI, MVAPICH, SLURM, and derived implementations. Support for MPICH was added in Kokkos 4.0
+
+5.2 Initialization by environment variable
+------------------------------------------
+
+Instead of using command-line arguments, one may use environment variables. The environment variables are identical to the arguments in Table 5.1 but they are upper case and the dash is replaced by an underscore. For example, if we want to set the number of threads to 3, we have
+
+  KOKKOS_NUM_THREADS=3
+
+5.3 Initialization by struct
+----------------------------
+
+Instead of giving `Kokkos::initialize() <../API/core/initialize_finalize/initialize.html>`_ command-line arguments, one may directly pass in initialization parameters using the `Kokkos::InitializationSettings` struct.  If one wants to set a options using the struct, one can use the set functions `set_xxx` where `xxx` is the identical to the arguments in Table 5.1 where the dash has been replaced by an underscore. To check if a variable has been set, one can use the `has_xxx` functions. Finally, to get the value that was set, one can use the `get_xxx` functions.
+
+
+If you do not set `num_threads`, Kokkos will try to determine a default value if possible or otherwise set it to 1. In particular, Kokkos can use the `hwloc` library to determine default settings using the assumption that the process binding mask is unique, i.e., that this process does not share any cores with another process. Note that the default value of each parameter is -1.
+
+Here is an example of how to use the struct.
+
+.. code-block:: cpp
+
+    Kokkos::InitializationSettings settings;
+    // 8 (CPU) threads
+    settinge.set_num_threads(8);
+    // If Kokkos was built with CUDA enabled, use the GPU with device ID 1.
+    settings.set_device_id(1);
+
+    Kokkos::initialize(settings);
+
+5.4 Finalization
+----------------
+
+At the end of each program, Kokkos needs to be shut down in order to free resources; do this by calling `Kokkos::finalize() <../API/core/initialize_finalize/finalize.html>`_. You may wish to set this to be called automatically at program exit, either by setting an `atexit` hook or by attaching the function to `MPI_COMM_SELF` so that it is called automatically at `MPI_Finalize`.
```

Comment:
I think the `atexit` or `MPI_COMM_SELF` thing is rather confusing. Perhaps we can split the paragraph to make visible that it is only an advanced use.